File: src/main/java/me/oriharel/machinery/inventory/InventoryNavigationItem.java
package me.oriharel.machinery.inventory;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

/**
 * a class for easy inventory navigation
 */
public class InventoryNavigationItem extends InventoryItem {

    protected String routeToName;
    protected Inventory parentInventory;

    public InventoryNavigationItem(String routeToName, Inventory parentInventory, int indexInInventory, Material material, int amount, String displayName,
                                   String... lore) {
        super(indexInInventory, material, amount, displayName, lore);
        this.routeToName = routeToName;
        this.parentInventory = parentInventory;
    }

    public InventoryNavigationItem(String routeToName, Inventory parentInventory, int indexInInventory, ItemStack itemStack) {
        super(indexInInventory, itemStack);
        this.routeToName = routeToName;
        this.parentInventory = parentInventory;
    }

    public InventoryNavigationItem(String routeToName, Inventory parentInventory, int indexInInventory, Material material, int amount, String displayName) {
        super(indexInInventory, material, amount, displayName);
        this.routeToName = routeToName;
        this.parentInventory = parentInventory;
    }

    public InventoryNavigationItem navigate() {
        parentInventory.navigateToNamedRoute(routeToName);
        return this;
    }
}

File: src/main/kotlin/me/oriharel/machinery/inventory/InventoryNavigationItem.kt
package me.oriharel.machinery.inventory

import org.bukkit.Material
import org.bukkit.inventory.ItemStack

open class InventoryNavigationItem : InventoryItem {
    private var routeToName: String?
    var parentInventory: Inventory

    constructor(routeToName: String?, parentInventory: Inventory, indexInInventory: Int, material: Material?, amount: Int, displayName: String?,
                vararg lore: String?) : super(indexInInventory, material, amount, displayName, *lore) {
        this.routeToName = routeToName
        this.parentInventory = parentInventory
    }

    constructor(routeToName: String?, parentInventory: Inventory, indexInInventory: Int, itemStack: ItemStack?) : super(indexInInventory, itemStack) {
        this.routeToName = routeToName
        this.parentInventory = parentInventory
    }

    constructor(routeToName: String?, parentInventory: Inventory, indexInInventory: Int, material: Material?, amount: Int, displayName: String?) : super(indexInInventory, material, amount, displayName) {
        this.routeToName = routeToName
        this.parentInventory = parentInventory
    }

    open fun navigate(): InventoryNavigationItem {
        parentInventory.navigateToNamedRoute(routeToName)
        return this
    }
}